azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/messager/AzureMessage.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.messager;
+
+import com.microsoft.azure.toolkit.lib.common.operation.IAzureOperation;
+import lombok.RequiredArgsConstructor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public abstract class AzureMessage implements IAzureMessage{
+    @Nonnull
+    public static AzureMessage.Context getContext() {
+        return getContext(IAzureOperation.current());
+    }
+
+    @Nonnull
+    public static AzureMessage.Context getContext(@Nullable IAzureOperation operation) {
+        return Optional.ofNullable(operation)
+                .map(o -> o.get(AzureMessage.Context.class, new AzureMessage.Context(operation)))
+                .orElse(new AzureMessage.Context(operation));
+    }
+
+    @RequiredArgsConstructor
+    public static class Context implements IAzureOperation.IContext {
```

Comment:
why we move Context from Messager -> Message?

Reply:
IMO. I think it would be easies to understand if it's similar to AzureTelemeter/AzureTelemetry

AzureMessage <-> AzureTelemetry, they manage contexts
AzureMessager <-> AzureTelemeter, they send/show message/telemetry data.